src/main/java/com/solvd/pages/SearchPage.java

```diff
@@ -3,28 +3,30 @@
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
-import org.openqa.selenium.support.PageFactory;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
-public class SearchPage {
-    WebDriver driver;
-
-    @FindBy(id = "prdocutname")
+public class SearchPage extends AbstractPage {
+    @FindBy(className = "prdocutname")
     List<WebElement> productNames;
 
     public SearchPage(WebDriver driver) {
-        this.driver = driver;
-        PageFactory.initElements(driver, this);
+        super(driver);
     }
 
-    public boolean areAllProductNamesMatch(String name) {
-        for (WebElement productName : productNames) {
-            if(!productName.getText().toLowerCase().contains(name.toLowerCase())) {
+
+    public boolean areAllProductNamesMatching(String name) {
```

Comment:
Consider validating if name is null or empty before processing to avoid NPEs or unintended behavior.